examples/file_browser.html

```diff
@@ -0,0 +1,119 @@
+        const NUM_DIR = 10;
+        const NUM_FILE = NUM_ROWS - NUM_DIR;
+
+        const COLUMN_NAMES = ['modified', 'writable'];
+
+        const formatter = new Intl.DateTimeFormat("en-us");
+
+        const _contents_cache = new Map();
+        function getContents(path, expand) {
+            // infinite recursive mock contents
+            let contents;
+            if (_contents_cache.has(path)) {
+                contents = _contents_cache.get(path);
+            } else {
+                contents = {path, modified: new Date(1000), writable: false};
+                _contents_cache.set(path, contents);
+            }
+
+            if (!expand || 'contents' in contents) {
+                return contents;
+            }
+
+            let sub = []
+            for (let i = 0; i < NUM_ROWS; i ++) {
+                sub.push({
+                    path: [...path, i < NUM_DIR ? `${i}/` : `${i}.txt`],
+                    modified: new Date(1000*86400*i),
+                    writable: false,
+                });
+            }
+
+            contents.contents = sub;
+            return contents;
+        }
+
+        function insertContents(rix, arr) {
+            const contents = getContents(arr[rix].path, true);
+            arr.splice(rix, 0, ...contents.contents);
+        }
+
+        function popContents(rix, arr) {
+            const {path} = arr[rix]
+            const contents = getContents(path, true);
+
+            // pop out the contents of the collapsed node and any expanded subnodes
+            let npop = contents.contents.length;
+            let check_ix = rix + 1 + npop;
+            while (arr[check_ix++].path.length > path.length) {
+                npop++;
+            }
+
+            arr.splice(rix + 1, npop);
+        }
+
+        let rows = getContents([], true).contents;
+
+        async function file_browser_model(start_col, start_row, end_col, end_row) {
+            const data = [];
+            for (let cix = start_col; cix < end_col; cix ++) {
+                const name = COLUMN_NAMES[cix];
+                data.push(rows.slice(start_row, end_row).map(c => {
+                    return name === 'modified' ? formatter.format(c[name]) : c[name];
+                }));
+            }
+
+            return {
+                num_rows: rows.length,
+                num_columns: COLUMN_NAMES.length,
+                column_indices: COLUMN_NAMES.map(col => [col]),
+                row_indices: rows.slice(start_row, end_row).map(c => c['path']),
+                data,
+            };
+        }
+
+        // for tree support
+        file_browser_model.collapse = rix => {
+            popContents(rix, rows);
+        };
+        file_browser_model.expand = rix => {
+            insertContents(rix + 1, rows);
+        }
+
+        const table = document.getElementsByTagName('regular-table')[0];
+        table.setDataModel(file_browser_model).then(() => {
+            table._view_cache.config = {
```

Comment:
Use `__config` to indicate this is legacy functionality bound for removal, ala `perspective.html` example.  This is not really how `row_pivot` parameter works, and is necessitates the `is_leaf` tweak below.

Reply:
I was trying to figure out a better way to pass in the name of the `row_indices` column. My proposal is that the `col`/`row_indices` arrays be replaced by Index objects (a la [Pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.html)) of the form:

```
{
  name: string;
  values: string[][];
}
```

Some simpler (but problematic) alternative proposals:
- expect an optional `.name` property on the existing `row_indices` array
- if `row_indices` is present, expect to find its name at `column_indices[0]` (ie `column_indices.length` should be `data.length + 1` when `row_indices` is present)

Reply:
I am now setting `row_pivots` via `__config`. However, right now `row_pivots` has no visible effect on the regular-table rendered by the `file_browser.html` example. The whole reason why I wanted to set `row_pivots` in the first place was to set the header text on the `row_indices` column, as implied here in the source:

https://github.com/jpmorganchase/regular-table/blob/ec0c5742f359170cd027fa901eb7a31b13003247/src/js/table.js#L84-L95

However, since `column_name` gets passed as the second arg of `header.draw`, that means it gets set as the column's `alias`, not as the actual displayed column name.

@texodus What's the distinction here between alias and actual column name? Is this something we can simplify for regular-table?